src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -495,10 +495,6 @@ public SolutionCompilationState RemoveAnalyzerReference(StateChange stateChange,
     public SolutionCompilationState WithProjectAnalyzerReferences(
         ProjectId projectId, IReadOnlyList<AnalyzerReference> analyzerReferences)
     {
-        var stateChange = this.Solution.WithProjectAnalyzerReferences(projectId, analyzerReferences);
-        if (stateChange.NewSolutionState == this.Solution)
-            return this;
-
         // The .Except() methods here aren't going to terribly cheap, but the assumption is adding or removing just the generators
         // we changed, rather than creating an entire new generator driver from scratch and rerunning all generators, is cheaper
         // in the end. This was written without data backing up that assumption, so if a profile indicates to the contrary,
```

Comment:
This comment should get moved in front of the Except calls; not sure if we missed that in the prior review.

Reply:
done.